DialogContentSettings remove scraper list and replace with a button that opens up a list

razzeee opened the pull request:
Here is a redesign of the scraper content settings dialog. Thanks to @mkortstiege @HitcherUK and @phil65 for contributing.
I was able to remove a lot of old now unneeded code.

![screenshot 2015-10-30 01 10 35](https://cloud.githubusercontent.com/assets/5943908/10835512/8706d066-7ea3-11e5-9178-1fc92344c5a1.png)

![screenshot 2015-10-30 01 10 44](https://cloud.githubusercontent.com/assets/5943908/10835511/870678b4-7ea3-11e5-9a42-9403510d6649.png)

razzeee committed:
DialogContentSettings:
Exchange list for a button + DialogSelect for scraper selection
Exchange content spinner for a button + DialogSelect for context selection

phil65 commented on addons/skin.confluence/720p/DialogContentSettings.xml:
```diff
@@ -60,158 +58,20 @@
-				<width>320</width>
-				<height>20</height>
-				<font>font12</font>
-				<label>$LOCALIZE[31312]</label>
```
Did you check if this label is now superfluous?

HitcherUK replied:
It now appears as the button label -
![screenshot004](https://cloud.githubusercontent.com/assets/133808/10840275/6908cec0-7ed5-11e5-8053-9871be5780a7.png)

phil65 replied:
ok, thx.

phil65 replied:
thinkin about this again.... shouldnt get the label get moved to core labels then, @Razzeee ?

Razzeee replied:
I'm not sure what I had there in my tests. But sure, we can do that, if you would like it.

MartijnKaijser commented:
looks great

MartijnKaijser commented:
We should change:
"Choose a scraper" to "Choose information provider"
"Scraper selection" to "Information provider selection"

ronie commented:
very, very nice :-)

Razzeee commented:
Updated the strings and moved the string to core as said by @phil65 
I also want to look into having a description set for each scraper in the selection window

Razzeee commented:
Replaced content spinner, somebody will need to realign the skin part as the header label is now removed due to redundancy.
![screenshot 2015-10-31 00 28 47](https://cloud.githubusercontent.com/assets/5943908/10860281/6ce5a158-7f66-11e5-84f2-367b486dc869.png)

jjd-uk commented:
Excellent, even better now with the spinner gone.

Razzeee commented:
Here are some updated screenshots:
![screenshot 2015-11-02 21 11 46](https://cloud.githubusercontent.com/assets/5943908/10893009/d552a3a2-81a6-11e5-93f2-09e1c2b09685.png)

One thing not everybody might like, all dialog lists are sorted alphabetically. There has been discussion about showing "None" at top, but I don't think it should have any special case. As it's rarely selected (if even at all) and music doesn't even has a "None" option.
![screenshot 2015-11-02 21 13 25](https://cloud.githubusercontent.com/assets/5943908/10893011/d55ca46a-81a6-11e5-8bac-40788e0e7431.png)
![screenshot 2015-11-02 21 14 14](https://cloud.githubusercontent.com/assets/5943908/10893010/d5596ebc-81a6-11e5-8942-d5ac378bd063.png)

Razzeee commented:
Thanks @tamland 
Please have a look again!

mkortstiege commented:
Nice cleanup. Thanks!

Razzeee commented:
@tamland is there any way to have a preselected item? Basically just a highlight?

tamland commented on xbmc/settings/dialogs/GUIDialogContentSettings.cpp:
```diff
@@ -89,46 +93,85 @@ bool CGUIDialogContentSettings::OnMessage(CGUIMessage &message)
+      else if (iControl == CONTROL_SCRAPER_LIST_BUTTON)
+      {
+        ADDON::TYPE type = ADDON::ScraperTypeFromContent(m_content);
+        std::string selectedAddonId;
```
Set it to the "old" id. the second parameter is in/out. it will highlight the ones you pass in.

Razzeee replied:
Ah, documentation just said out. Thank you. 
Btw I think the name of the method is not very self explanatory. Any chance to change it to something better?

tamland replied:
I guess, if you can change it to something *better*:) I didn't write it, but it sound ok enough to me. SelectAddonID: opens a dialog for selecting addon..

Razzeee replied:
Well I think it's missing the open dialog part and the ID part is kind of misleading also? Probably a function that evolved to far more, then it originally was?

Hitcher committed:
[Confluence] Update Content Settings dialog

Razzeee commented:
Rebased and squashed.
I'm done with it, please review some more guys. We can merge this, when the next version opens up.

HitcherUK commented:
The only other thing I would personally like happen with this dialog is to remove the 3 separate buttons (20, 21, 22) and the heading label into the default grouplist (5) and have them handled in core like the other buttons.
@phil65 @ronie @BigNoid ?

ronie commented:
yup, my thoughts exactly.

MartijnKaijser commented:
design wise +1 :)

phil65 commented:
yup, what @HitcherUK said. That step would be needed if we want to merge the dialog with similar ones in the future.

Razzeee commented:
Well, I don't have the knowlegde for that, so I need a pointer in the right direction, but should probably be another PR anyway.

Razzeee commented:
jenkins build this please
